### PR TITLE
Internal cache impl for SharedBlobCacheService

### PR DESCRIPTION
**Pull request by romseygeek**
This pulls all LFU details into an internal cache implementation behind a simple
cache interface with just three methods - get(), evictMatching() and close().  It
is a first step to adding an alternative LRU implementation.

### CODE REVIEW

**Comment by romseygeek** [comment]
Take 2, as a replacement for #99625 which was trying to be too clever.  This is a much simpler straight refactoring of all the LFU caching methods behind an interface.  The only slightly tricky part is the addition of an intermediate CacheEntry object with a touch() method, which allows us to keep the caching of the last accessed region.

File: x-pack/plugin/blob-cache/src/main/java/org/elasticsearch/blobcache/shared/SharedBlobCacheService.java [hunk 1]
```diff
@@ -1021,18 +775,16 @@ public boolean tryRead(ByteBuffer buf, long offset) throws IOException {
...
-                }
+                fileRegion.touch();
+
             } else {
-                fileRegion = get(cacheKey, length, startRegion);
+                fileRegion = cache.get(cacheKey, length, startRegion);
             }
-            final var region = fileRegion.chunk;
-            if (region.tracker.checkAvailable(end - getRegionStart(startRegion)) == false) {
+            final var region = fileRegion;
```
Comment (original-brownbear) [inline x-pack/plugin/blob-cache/src/main/java/org/elasticsearch/blobcache/shared/SharedBlobCacheService.java:783]: Why change this? :) Seems we're increasing the number of field accesses here, not sure it matters much but seems unnecessary?

**Review by original-brownbear — APPROVED** [review]
LGTM this took so long. Just one question but that doesn't need another review :)

File: x-pack/plugin/blob-cache/src/main/java/org/elasticsearch/blobcache/shared/SharedBlobCacheService.java (hunk 1 above)
Comment (romseygeek) [reply x-pack/plugin/blob-cache/src/main/java/org/elasticsearch/blobcache/shared/SharedBlobCacheService.java]: Think that was a leftover from a previous version, I've changed it back again

**Pull request merged**